src/DataLayouts/DataLayouts.jl

```diff
@@ -411,6 +415,100 @@ end
     IFH{SS, Ni}(dataview)
 end
 
+# ======================
+# Data0D DataLayout
+# ======================
+
+Base.size(data::Data0D) = (1, 1, 1, 1, 1)
+
+struct F{S, A} <: Data0D{S}
+    array::A
+end
+
+function F{S}(array::AbstractArray{T, 2}) where {S, T}
+    @assert size(array, 1) == 1
+    check_basetype(T, S)
+    F{S, typeof(array)}(array)
+end
+
+function F{S}(array::AbstractVector{T}) where {S, T}
+    @assert size(array, 1) == 1
+    @assert typesize(T, S) == 1
+    F{S}(reshape(array, (:, 1)))
+end
+
+function F{S}(ArrayType) where {S}
+    T = eltype(ArrayType)
+    F{S}(ArrayType(undef, 1, typesize(T, S)))
+end
+
+# @inline function Base.getproperty(data::VF{S}, i::Integer) where {S}
+#     array = parent(data)
+#     T = eltype(array)
+#     SS = fieldtype(S, i)
+#     offset = fieldtypeoffset(T, S, i)
+#     nbytes = typesize(T, SS)
+#     dataview = @inbounds view(array, :, (offset + 1):(offset + nbytes))
+#     VF{SS}(dataview)
+# end
+@inline function Base.getproperty(data::F{S}, i::Integer) where {S}
+    array = parent(data)
+    T = eltype(array)
+    SS = fieldtype(S, i)
+    offset = fieldtypeoffset(T, S, i)
+    nbytes = typesize(T, SS)
+    dataview = @inbounds view(array, :, (offset + 1):(offset + nbytes))
+    F{SS}(dataview)
+end
+
+# @inline function Base.getindex(data::VF{S}, v::Integer) where {S}
+#     @boundscheck 1 <= v <= size(parent(data), 1) ||
+#                  throw(BoundsError(data, (v,)))
+#     dataview = @inbounds view(parent(data), v, :)
+#     @inbounds get_struct(dataview, S)
+# end
+@inline function Base.getindex(data::F{S}, v::Integer) where {S}
+    @boundscheck 1 <= v <= size(parent(data), 1) ||
+                 throw(BoundsError(data, (v,)))
+    dataview = @inbounds view(parent(data), v, :)
+    @inbounds get_struct(dataview, S)
+end
```

Comment:
The folllowing should be sufficient:
```suggestion
@inline function Base.getindex(data::F{S) where {S}
    @inbounds get_struct(parent(data), S)
end
```

Reply:
Similarly for `Base.setindex!`?

Reply:
Yes